src/microbe_stage/MicrobeHUD.cs

```diff
@@ -247,7 +247,10 @@ public override void ApplySpeedMode(bool fastModeEnabled)
             return;
         }
 
-        stage.WorldSimulation.WorldTimeScale = fastModeEnabled ? 2 : 1;
+        var resultingModifier = fastModeEnabled ? 2 : 1;
+
+        stage.WorldSimulation.WorldTimeScale = resultingModifier;
+        UpdateProcessPanelExternalSpeedModifier(resultingModifier);
```

Comment:
I think calling this here is not enough, because the speed mode is preserved when making and loading a save, so the process speed actually needs to be checked and applied constantly, otherwise the GUI state might get desynced from the real simulation state when loading a save.